R/utils-io.R

```diff
@@ -1,11 +1,19 @@
-# readLines <- function(...) {
-#   drive_abort("In this house, we use ??? for UTF-8 reasons.")
-# }
+readLines <- function(...) {
+  drive_abort("In this house, we use{.fun read_utf8} for UTF-8 reasons.")
+}
 
 writeLines <- function(...) {
   drive_abort("In this house, we use {.fun write_utf8} for UTF-8 reasons.")
 }
 
+# https://github.com/gaborcsardi/rencfaq#with-base-r
+read_utf8 <- function (path) {
+  opts <- options(encoding = "native.enc")
+  withr::defer(options(opts))
+  x <- base::readLines(path, encoding = "UTF-8", warn = FALSE)
+  x
+}
+
```

Comment:
Needing to read UTF-8 files with non-ASCII characters in the tests for this PR provided the nudge for me to complete the in-house `read_utf8()` / `write_utf8()` pair.

Reply:
Not sure if you want to convert newlines automatically or not, but this currently open a text connection so the newlines are going to be normalized to the OSs default.

It also is going to have problems if the file path is not representable in the native encoding.

Reply:
The current simplistic implementation of these functions seems to be sufficient for their usage so far. But it's good to have advance warning re: where they are going to fall short. If a user bumps into this, I'll take that as a sign that it's time to use brio.